backend/src/nodes/image_nodes.py

```diff
@@ -155,22 +155,37 @@ def run(
         img = (np.clip(img, 0, 1) * 255).round().astype("uint8")
 
         os.makedirs(base_directory, exist_ok=True)
-
-        # Write image with opencv if path is ascii, since imwrite doesn't support unicode
-        # This saves us from having to keep the image buffer in memory, if possible
-        if full_path.isascii():
-            status = cv2.imwrite(full_path, img)
+        # Any image not supported by cv2, will be handled by pillow.
+        if extension not in ["png", "jpg", "gif", "tiff", "webp"]:
+            status = 1  # spoof
+            channels = get_h_w_c(img)[2]
+            # if array is BGRA or BGR it must be converted to RGBA or RGB, any grayscale should pass through fine.
+            # if it has neither 4, 3, nor 1 channels we can't use pillow.
+            if channels not in [4, 3, 1]:
+                raise RuntimeError(f"Unsupported number of channels. Saving .{extension} images is only supported for "
+                                   f"grayscale, RGB, and RGBA images.")
+            elif channels == 3:
+                img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
+            elif channels == 4:
+                img = cv2.cvtColor(img, cv2.COLOR_BGRA2RGBA)
```

Comment:
I don't think this is optimal. This kinda feels like bike-shedding, but there are 3 things I don't like about this:
1. You needed to comment that single-channel images are passed through, when that should be written as code instead. IMO, comments shouldn't say what the program does (code should do that), but why the program does things.
2. There are now 2 places in code that check for RGB and RGBA images.
3. The `if channels not in [4, 3, 1]` needs to know all supported channel formats.

I would suggest a simple if-elif chain.

```suggestion
            if channels == 1:
                # PIL supports grayscale images just fine, so we don't need to do any conversion
                pass
            elif channels == 3:
                img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
            elif channels == 4:
                img = cv2.cvtColor(img, cv2.COLOR_BGRA2RGBA)
            else:
                raise RuntimeError(f"Unsupported number of channels. Saving .{extension} images is only supported for "
                                   f"grayscale, RGB, and RGBA images.")
```

This makes it obvious why we don't do a conversion for single-channel images, we only check each channel count once, and the error naturally appears in the `else` branch.